gr-digital/lib/constellation_encoder_bc_impl.cc

```diff
@@ -0,0 +1,57 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2020 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "constellation_encoder_bc_impl.h"
+#include <gnuradio/io_signature.h>
+
+namespace gr {
+namespace digital {
+
+constellation_encoder_bc::sptr
+constellation_encoder_bc::make(constellation_sptr constellation)
+{
+    return gnuradio::get_initial_sptr(new constellation_encoder_bc_impl(constellation));
+}
+
+constellation_encoder_bc_impl::constellation_encoder_bc_impl(
+    constellation_sptr constellation)
+    : sync_interpolator("constellation_encoder_bc",
+                        io_signature::make(1, 1, sizeof(unsigned char)),
+                        io_signature::make(1, 1, sizeof(gr_complex)),
+                        constellation->dimensionality()),
+      d_constellation(constellation),
+      d_dim(constellation->dimensionality())
```

Comment:
i'm wondering if the dimensionality > 1 is ever used with the constellation objects:
```c++
void constellation::map_to_points(unsigned int value, gr_complex* points)
{
    for (unsigned int i = 0; i < d_dimensionality; i++)
        points[i] = d_constellation[value * d_dimensionality + i];
}
```
i would think this would be `points[i] = d_constellation[i * d_dimensionality + value];` to ever work at all

anyhow i'm going to leave out the test with dimensionality >1 until i can figure out whether dim>1 works at all

Reply:
I haven't seen dimensionality been used ever. I'm bound to think this is a concept someone once had but never finished to implement all aspects of it.